actionpack/lib/action_dispatch/journey/nodes/node.rb

```diff
@@ -32,7 +32,7 @@ def to_sym
         end
 
         def name
-          left.tr "*:".freeze, "".freeze
+          -(left.tr "*:", "")
```

Comment:
What is going on here? i'm not familiar with `-()` invocation, is that what's saving memory?

Reply:
http://ruby-doc.org/core-2.5.0/String.html#method-i-2D-40.

It is equivalent to `left.tr("*:", "").freeze`, except that it only freeze if it is not frozen already.

Reply:
oh... you got to read https://samsaffron.com/archive/2018/02/16/reducing-string-duplication-in-ruby carefully, and https://bugs.ruby-lang.org/issues/14478 ... left.tr("*:", "").freeze will not de-duplicate in Ruby 2.5 ... -(left.tr("*:", "")) will ... and amusingly -(left.tr("*:", "").freeze) until my bug is fixed and backported, which is a 3 line change ;)